### PR TITLE
Http status command randomness

### LINKED ISSUE #731: Random Cat Or Dog Pic When Not Specified In .http_status
## Description

When not specified there should be a random dog or cat pic response.
EX: `.status cat ` and `.status dog` would be unchanged but if `.status 404` was entered it would choose dog or cat randomly.

## Reasoning

This would be useful if you dont care about the picture and are using the command to find the meaning of the status code.
I use it for this reason mainly. You just dont need to know/spend time typing an extra argument when you dont care.

## Proposed Implementation

Change the group command, instead of directing to help command, pick one of the 2 subcommands randomly and run it with \_\_call__()

## Would you like to implement this yourself?

- [x] I'd like to implement this feature myself
- [ ] Anyone can implement this feature

### PR DESCRIPTION
**Pull request by Icebluewolf**
## Relevant Issues

#731 

Closes #731 

## Description

When cat or dog subcommands are not invoked the parent command (http_status) will show a cat or dog picture, relating to the status code given, randomly. 

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

File: bot/exts/evergreen/status_codes.py [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
         if not ctx.invoked_subcommand:
-            await invoke_help_command(ctx)
+            random = randint(0, 1)
+            if random == 0:
+                subcmd = self.bot.get_command("http_status cat")
+            else:
+                subcmd = self.bot.get_command("http_status dog")
+
+            if await commands.Group.can_run(subcmd, ctx):
+                await subcmd(ctx, code)
```
Comment (ToxicKidz) [inline bot/exts/evergreen/status_codes.py:31]:
```suggestion
            if await subcmd.can_run(ctx):
                await subcmd(ctx, code)
```

File: bot/exts/evergreen/status_codes.py [hunk 2]
```diff
@@ -1,25 +1,34 @@
...
+    """
 
     def __init__(self, bot: commands.Bot):
         self.bot = bot
 
-    @commands.group(name="http_status", aliases=("status", "httpstatus"))
-    async def http_status_group(self, ctx: commands.Context) -> None:
+    @commands.group(name="http_status", aliases=("status", "httpstatus"), invoke_without_command=True)
+    async def http_status_group(self, ctx: commands.Context, code: int) -> None:
         """Group containing dog and cat http status code commands."""
```
Comment (ToxicKidz) [inline bot/exts/evergreen/status_codes.py:22]: This docstring may need to be updated.

**Review by ToxicKidz — CHANGES_REQUESTED** [review]

**Comment by Icebluewolf** [comment]
I think I fixed all the issues. Sorry for the troubles. When I committed it went to the wrong branch so I had to revert it and that deleted my code on my PC somehow. So I had to redo it and must have forgot to lint.

File: bot/exts/evergreen/status_codes.py [hunk 3]
```diff
@@ -1,25 +1,34 @@
...
-        if not ctx.invoked_subcommand:
-            await invoke_help_command(ctx)
+    @commands.group(name="http_status", aliases=("status", "httpstatus"), invoke_without_command=True)
+    async def http_status_group(self, ctx: commands.Context, code: int) -> None:
+        """Randomize cat/dog picture output when a sub-command is not specified."""
+        subcmd = choice((self.http_cat, self.http_dog))
+        # if random == 0:
+        #     subcmd = self.bot.get_command("http_status cat")
+        # else:
+        #     subcmd = self.bot.get_command("http_status dog")
```
Comment (ToxicKidz) [inline bot/exts/evergreen/status_codes.py:28]: These comments should be deleted

**Review by ToxicKidz — CHANGES_REQUESTED** [review]

File: bot/exts/evergreen/status_codes.py [hunk 4]
```diff
@@ -1,25 +1,34 @@
...
 HTTP_CAT_URL = "https://http.cat/{code}.jpg"
 
 
 class HTTPStatusCodes(commands.Cog):
-    """Commands that give HTTP statuses described and visualized by cats and dogs."""
+    """
+    Commands that give HTTP statuses described and visualized by cats and dogs.
+
+    Give a cat or dog visualization randomly if the option is not filled.
+    """
```
Comment (jb3) [inline bot/exts/evergreen/status_codes.py:16]: I think we can tidy up these docs a little.
```suggestion
    """
    Fetch an image depicting HTTP status codes as a dog or a cat.

    If neither animal is selected a cat or dog is chosen randomly for the given status code.
    """
```

File: bot/exts/evergreen/status_codes.py [hunk 5]
```diff
@@ -1,25 +1,34 @@
...
         self.bot = bot
 
-    @commands.group(name="http_status", aliases=("status", "httpstatus"))
-    async def http_status_group(self, ctx: commands.Context) -> None:
-        """Group containing dog and cat http status code commands."""
-        if not ctx.invoked_subcommand:
-            await invoke_help_command(ctx)
+    @commands.group(name="http_status", aliases=("status", "httpstatus"), invoke_without_command=True)
+    async def http_status_group(self, ctx: commands.Context, code: int) -> None:
+        """Randomize cat/dog picture output when a sub-command is not specified."""
```
Comment (jb3) [inline bot/exts/evergreen/status_codes.py:23]:
```suggestion
        """Choose a cat or dog randomly for the given status code."""
```

**Review by jb3 — CHANGES_REQUESTED** [review]
Some documentation amendments.

File: bot/exts/evergreen/status_codes.py (hunk 3 above)
Comment (Icebluewolf) [inline bot/exts/evergreen/status_codes.py:28]:
```suggestion

```

**Review by ToxicKidz — APPROVED** [review]
Thanks, LGTM! :shipit:

**Review by doublevcodes — COMMENTED** [review]
Looks good! 👍🏾

**Review by Kronifer — APPROVED** [review]
No problem here, LGTM!

File: bot/exts/evergreen/status_codes.py [hunk 6]
```diff
@@ -1,25 +1,30 @@
...
-    async def http_status_group(self, ctx: commands.Context) -> None:
-        """Group containing dog and cat http status code commands."""
-        if not ctx.invoked_subcommand:
-            await invoke_help_command(ctx)
+    @commands.group(name="http_status", aliases=("status", "httpstatus"), invoke_without_command=True)
+    async def http_status_group(self, ctx: commands.Context, code: int) -> None:
+        """Choose a cat or dog randomly for the given status code."""
+        subcmd = choice((self.http_cat, self.http_dog))
+
+        if await subcmd.can_run(ctx):
```
Comment (RohanJnr) [inline bot/exts/evergreen/status_codes.py:26]: Everything looks good 

except for the
`command,can_run()` if statement., if the user can the `group` main command, then the subcommands should have no problem running. We don't have any checks on the functions too. The invalid argument error in any way
caught in the error handler. But I guess it doesn't hurt to have it.

**Review by RohanJnr — APPROVED** [review]

File: bot/exts/evergreen/status_codes.py (hunk 6 above)
Comment (Icebluewolf) [reply bot/exts/evergreen/status_codes.py:26]: I was told to use this instead of \_\_call__ so that check would run.

Comment (ToxicKidz) [reply bot/exts/evergreen/status_codes.py:26]: Right, but this group command has the same checks as the subcommands (which is the global check) so there isn't really any point in calling `can_run`.

File: bot/exts/evergreen/status_codes.py [hunk 7]
```diff
@@ -1,25 +1,30 @@
...
+            await subcmd(ctx, code)
```
Comment (Icebluewolf) [inline bot/exts/evergreen/status_codes.py:29]:
```suggestion
        await subcmd(ctx, code)
```

**Review by ChrisLovering — APPROVED** [review]

**Pull request merged**